gui/src/main.rs

```diff
@@ -55,33 +62,44 @@ impl Default for GuiConfig {
                     String::from("no-cache"),
                 ),
             ])),
-            response: None,
+            import_window_open: false,
+            import_file_path: String::from(""),
+            import_mode: ImportMode::COLLECTION,
         }
     }
 }
 
 pub struct Gui {
-    pub config: GuiConfig,
-    pub rt: runtime::Runtime,
+    pub config: Arc<RwLock<GuiConfig>>,
```

Comment:
Arc is atomically reference counted, essentially letting multiple threads have access to this memory.

RwLock is a Read-Write Lock. It lets multiples difference owners have read access, but only one can have write access at a time.